### DISCUSSION TITLE
Are performance tests something that anyone uses now

### DISCUSSION THREAD

**Discussion by akurtakov** [eclipse-platform/eclipse.platform.releng.aggregator#828] [General]
As part of https://github.com/eclipse-pde/eclipse.pde/pull/445#issuecomment-1406334643 it is visible that PerformanceTestCase leaked in non-performance tests but this raised the even bigger question. Are these perf tests usable for anything? 
Every I-build has a page like https://download.eclipse.org/eclipse/downloads/drops4/I20230125-1800/performance/performance.php if they manage to complete.
For the record these tests were not produced for multiple releases and @SDawley revived them last release but never had the time to take them to the state where all of them complete. 
Monitoring results and investigating regressions would be nice but right now is beyond the time people involved in releng can spend thus we are letting these to run "as they do so, until the do so" as it's time spend for no benefit. If anyone wants to step up for that work he is more than welcome. 
P.S. It has been identified that what offered today by PeformanceTestCase should today be a JUnit Rule maybe even something build in https://junit.org/junit4/javadoc/4.12/org/junit/rules/Stopwatch.html but even porting the tests is a huge investment which doesn't make sense if performance not monitored regularly.

**Comment by iloveeclipse** [comment]
> Are these perf tests usable for anything?

If the question would be "Are these perf tests usable for anything **now**?" the answer would be clear "**no**".
They **were** usable long time ago. 

As long as we don't have resources to monitor the fails, they aren't useful, at least I haven't seen any bug report referring to the performance failure in those tests for years.

- If they hinder us to do platform maintenance, we could remove them.
- If they are just "lying around" and do not harm, we could keep them.

Every time seeing the part of the glorious days of Eclipse 3.x going away makes me sad.
But reality is that with remaining resources we can't afford to keep on development level that previous Eclipse developers generation left for us.

**Comment by akurtakov** [comment]
I'll stop running the tests as they take the precious resources needed to execute tests and compiling the native bits - they all run on the same JIPP instance now.
